src/internal/storage/kv/kv.go

```diff
@@ -0,0 +1,40 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"hash/crc64"
+)
+
+// ErrKeyNotFound is returned when the store does not contain a key
+var ErrKeyNotFound = errors.New("key not found")
+
+// ValueCallback is the type of functions used to access values
+type ValueCallback = func([]byte) error
+
+// GetPut supports the basic GetF and Put operations
+type GetPut interface {
+	Put(ctx context.Context, key, value []byte) error
+	GetF(ctx context.Context, key []byte, cb ValueCallback) error
```

Comment:
Could this just be `Get`? We have been moving away from the `F` suffix for callback based functions in other parts of the codebase. Also, really minor nit, but I think this interface should be `PutGet` rather than `GetPut`.

Reply:
Yeah it can be `Get` instead

I don't know where you're getting `PutGet`
In the `io` package, there is `io.ReadWriter` so any analogy of Read -> Get and Put -> Write has that convention.
There is also just concatenating them in alphabetically order, which supports `GetPut`.

Reply:
Yeah, that is a fair point. I was mostly pulling that from the way I think about it (you write a chunk before you can get it) and because you laid out the functions with `Put` before `Get`. I agree we should follow the Go conventions, so if you just make the `GetF` -> `Get` change and list `Get` before `Put` then we should be good to go.